### ISSUE TITLE
param! :foo, ActionDispatch::Http::UploadedFile will throw an exception when nil

### ISSUE THREAD

**Issue by ccdredzik** [nicolasblanco/rails_param#80]
## Basic Info

* rails_param Version: 1.0.2
* Ruby Version: 3.0.2
* Rails Version: 6.1.4.1

## Issue description

I have an endpoint that has the following param:

```
param! :foo, ActionDispatch::Http::UploadedFile
```

When sending foo: nil it will produce the following error:

```
'' is not a valid ActionDispatch::Http::UploadedFile
```

This was not a problem with 0.11.2 version of the rails_param gem.

Stack trace:

```
      "/Users/ak/.rvm/gems/ruby-3.0.2/gems/rails_param-1.0.2/lib/rails_param/param_evaluator.rb:82:in `rescue in coerce'",
      "/Users/ak/.rvm/gems/ruby-3.0.2/gems/rails_param-1.0.2/lib/rails_param/param_evaluator.rb:76:in `coerce'",
      "/Users/ak/.rvm/gems/ruby-3.0.2/gems/rails_param-1.0.2/lib/rails_param/param_evaluator.rb:17:in `param!'",
      "/Users/ak/.rvm/gems/ruby-3.0.2/gems/rails_param-1.0.2/lib/rails_param/param.rb:3:in `param!'",
      "/Users/ak/Projects/project/app/controllers/a_controller.rb:34:in `update'",
```

**Comment by ccdredzik** [comment]
It will produce the same issue if you set the type to `Array` but send a `nil`. It will work just fine if you set it to `Hash` but send a `nil`. I would expect that unless `required: true`, `nil` is always nice.

**Comment by iMacTia** [comment]
Hi @ccdredzik and thanks for opening this issue!
This looks somewhat related to #43.
When you say "When sending `nil`" do you mean you're explicitly sending `nil` as a value, or are you simply omitting the parameter?

**Comment by ccdredzik** [comment]
@iMacTia sorry for the late reply. If the param is set to type `Array` we can reproduce this by sending a `PATCH` or `POST` with a `Content-Type: application/json` and a JSON like:

```
{ foo: null }
```

For `ActionDispatch::Http::UploadedFile`  we can reproduce this by sending a `PATCH` with a `Content-Type: multipart/form-data` and no param.

**Comment by ccdredzik** [comment]
@iMacTia made a PR to demonstrate the issue

**Comment by iMacTia** [comment]
Thanks for the additional details and the PRs!
The issue when sending `null` explicitly is up for debate, but I agree the latter case is surprising.
I'll give this a look as soon as possible.
@ifellinaholeonce feel free to chime in if you have any time/thoughts on this 👍

**Comment by ccdredzik** [comment]
@iMacTia @ifellinaholeonce I have just updated my PR with commentary. Please review in a spare moment.

**Comment by ccdredzik** [comment]
@iMacTia For the record. I don't think there is much debate about it. Sending null explicitly is a completely normal and sane behaviour IMHO and should be handled without exceptions. Otherwise if I deploy an API with this gem, every single consumer of my API will have to build integration, where every optional value is being checked for presence and omitted if null. That simply doesn't make any sense, and is a huge waste of everyone's time.

**Comment by ccdredzik** [comment]
For readability #81 is now only fixing an issue in handling strings in hashes that I've found surfacing while fixing the root cause of this issue in #82.

**Comment by iMacTia** [comment]
@ccdredzik can we consider this issue closed by #82 ?

**Comment by ccdredzik** [comment]
@iMacTia please do
